File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l = 0
        r = len(numbers)-1
        while l!=r:
            total = numbers[l]+numbers[r]
            if(total==target):
                return [l+1,r+1]
            else:
                if total > target:
                    r = r-1
                else:
                    l = l+1

        return [-1] 
            

File: Data Structures & Algorithms/car-fleet/submission-0.cpp
class Solution {
public:
    int carFleet(int target, vector<int>& position, vector<int>& speed) {
        vector<pair<int,int>> carFleet(speed.size());
        stack<double> timeSt;
        int result = 0;
        for(int i = 0;i<speed.size();i++){
            carFleet[i]=(pair<int,int>{position[i],speed[i]});
        }
        sort(carFleet.begin(), carFleet.end(),[](const auto& a, const auto& b){
            return a.first > b.first;
        });

        for(int i=0;i<carFleet.size();i++){
            double time = (double)(target - carFleet[i].first) / carFleet[i].second;
            if(timeSt.empty() || time>timeSt.top()){
                timeSt.push(time);
            }
        }

        return timeSt.size();
    }
};
